src/Middleware/HttpLogging/src/HttpLoggingMiddleware.cs

```diff
@@ -84,36 +133,89 @@ private async Task InvokeInternal(HttpContext context)
 
                 if (options.LoggingFields.HasFlag(HttpLoggingFields.RequestProtocol))
                 {
-                    AddToList(list, nameof(request.Protocol), request.Protocol);
+                    if (httpEnabled)
+                    {
+                        AddToList(list, nameof(request.Protocol), request.Protocol);
+                    }
+                    if (w3cEnabled)
+                    {
+                        AddToList(w3cList, nameof(request.Protocol), request.Protocol);
+                    }
                 }
 
                 if (options.LoggingFields.HasFlag(HttpLoggingFields.RequestMethod))
                 {
-                    AddToList(list, nameof(request.Method), request.Method);
+                    if (httpEnabled)
+                    {
+                        AddToList(list, nameof(request.Method), request.Method);
+                    }
+                    if (w3cEnabled)
+                    {
+                        AddToList(w3cList, nameof(request.Method), request.Method);
+                    }
                 }
 
-                if (options.LoggingFields.HasFlag(HttpLoggingFields.RequestScheme))
+                if (httpEnabled && options.LoggingFields.HasFlag(HttpLoggingFields.RequestScheme))
                 {
                     AddToList(list, nameof(request.Scheme), request.Scheme);
                 }
 
-                if (options.LoggingFields.HasFlag(HttpLoggingFields.RequestPath))
+                if (httpEnabled && options.LoggingFields.HasFlag(HttpLoggingFields.RequestPath))
                 {
                     AddToList(list, nameof(request.PathBase), request.PathBase);
                     AddToList(list, nameof(request.Path), request.Path);
                 }
 
-                if (options.LoggingFields.HasFlag(HttpLoggingFields.RequestQuery))
+                if (httpEnabled && options.LoggingFields.HasFlag(HttpLoggingFields.RequestQueryString))
                 {
                     AddToList(list, nameof(request.QueryString), request.QueryString.Value);
                 }
 
+                if (w3cEnabled && options.LoggingFields.HasFlag(HttpLoggingFields.RequestQuery))
+                {
+                    // TODO - query is written as a list of {Key}:{Value} pairs delimited by semicolon -
+                    // is there a better/standardized format?
```

Comment:
Can you just [enable this in IIS](https://docs.microsoft.com/en-us/iis/configuration/system.applicationhost/sites/sitedefaults/logfile/#configuration) and see what it does? The [spec](https://www.w3.org/TR/WD-logfile.html) says the `uri-query` is a "Query portion alone of URI, field has type \<uri\>". I assume they just want `request.QueryString`.

We should make sure the logs this middleware produces is a close to IIS's W3C logs as possible unless we know IIS is doing something wrong.